controllers/operandrequest/reconcile_operator.go

```diff
@@ -54,18 +55,22 @@ func (r *Reconciler) reconcileOperator(ctx context.Context, requestInstance *ope
 		registryKey := requestInstance.GetRegistryKey(req)
 		registryInstance, err := r.GetOperandRegistry(ctx, registryKey)
 		if err != nil {
-			if apierrors.IsNotFound(err) {
-				r.Recorder.Eventf(requestInstance, corev1.EventTypeWarning, "NotFound", "NotFound OperandRegistry NamespacedName %s", registryKey.String())
-				klog.Errorf("failed to find OperandRegistry %s : %v", registryKey.String(), err)
-				requestInstance.SetNotFoundOperatorFromRegistryCondition(registryKey.String(), operatorv1alpha1.ResourceTypeOperandRegistry, corev1.ConditionTrue)
-				mergePatch, _ := json.Marshal(map[string]interface{}{
-					"metadata": map[string]interface{}{
-						"annotations": map[string]interface{}{
-							constant.FindOperandRegistry: "true",
-						},
+			r.Recorder.Eventf(requestInstance, corev1.EventTypeWarning, "NotFound", "NotFound OperandRegistry NamespacedName %s", registryKey.String())
+			klog.Errorf("failed to find OperandRegistry %s : %v", registryKey.String(), err)
+			requestInstance.SetNotFoundOperatorFromRegistryCondition(registryKey.String(), operatorv1alpha1.ResourceTypeOperandRegistry, corev1.ConditionTrue)
```

Comment:
we didn't check `apierrors.IsNotFound(err)`, why here still use Set**NotFoundOperator**FromRegistryCondition?

Reply:
I think when there is a error happened, no matter it is a `not found` error or others, the result is always the same that we failed to get the `OperandRegistry`.